kernel/device.c

```diff
@@ -46,6 +48,7 @@ void z_sys_device_do_config_level(s32_t level)
 		__device_init_end,
 	};
 
+	z_sys_device_level = level;
```

Comment:
Why is this done here and not at the exit of the function?

Reply:
Because if it's processing `POST_KERNEL` initialization then then it's post kernel.